examples/scripts/project/categorization/bootstrap_taxonomy.py

```diff
@@ -0,0 +1,169 @@
+
+    """
+
+    # Create the tamr client
+    tamr_client = tbox.utils.client.create(**instance_connection_info)
+
+    # Retrieve the project
+    project = tamr_client.projects.by_resource_id(categorization_project_id)
+    LOGGER.info(f"Retrieved project with name: {project.name}")
+
+    # Get the project taxonomy
+    # NOTE: .taxonomy() function breaks on newer tamr instances SUP-5075
+    try:
+        project.taxonomy()
+    except requests.exceptions.RequestException:
+        no_taxonomy_error = f"Project {project.name} is not associated with any taxonomy yet."
+        raise RuntimeError(no_taxonomy_error)
+    LOGGER.info(f"Retrieved project taxonomy with name: {project.taxonomy().name}")
+
+    # Bootstrap all available categories
+    categories = project.taxonomy().categories()
+    category_list = []
+    for category in categories:
+        category_list.append(category.path)
+    # For bootstrapping at a specific tier = -1 (leaf), 1, 2, ... use the existing function
+    # category_set = metrics._get_categories_at_tier(project=project, tier=1)
+    # category_list = [category.split("|") for category in category_set]
+    category_list.sort()
+
+    # Create a dataset with taxonomy categories
+    dataset_name = f"{project.unified_dataset().name}_taxonomy_bootstrap_dataset"
+    try:
+        project.client.datasets.by_name(dataset_name)
+    except KeyError:
+        # Dataset with `dataset_name` does not exist in Tamr. Proceed with dataset creation.
+        pass
+    else:
+        dataset_exists_error = (
+            f"A dataset with name {dataset_name} already exists. Try again after deleting the "
+            "dataset."
+        )
+        LOGGER.error(dataset_exists_error)
+        raise RuntimeError(dataset_exists_error)
+
+    # Create a dictionary of full path between string and list (used as label path)
+    taxonomy_dict = {", ".join(category): category for category in category_list}
+
+    # Create a dataframe
+    df = pd.DataFrame(list(taxonomy_dict.items()), columns=COLUMN_NAMES)
+
+    # Add category data and hash for tamr_id
+    df[CATEGORY_ATTRIBUTE_NAME] = df[FULL_PATH_LIST_NAME].apply(lambda x: x[-1])
+    df[PK_NAME] = df[FULL_PATH_STR_NAME].apply(lambda x: hash(x))
+    df.drop(FULL_PATH_LIST_NAME, axis=1, inplace=True)
+
+    # Create a dataset in Tamr
+    taxonomy_dataset = project.client.datasets.create_from_dataframe(
+        df, primary_key_name=PK_NAME, dataset_name=dataset_name
+    )
+    LOGGER.info(f"Created a dataset in Tamr with name: {taxonomy_dataset.name}")
+
+    # Add the dataset into the project
+    project.add_input_dataset(taxonomy_dataset)
+    LOGGER.info(f"Added {taxonomy_dataset.name} to project {project.name}")
+
+    # Create a unified attribute to map taxonomy categories
+    unified_attribute_name = "Taxonomy Categories"
+    assert schema.create_unified_attribute(project, unified_attribute_name=unified_attribute_name)
```

Comment:
Should this be mapped to an existing attribute? Otherwise the learning on this attribute only applies to the bootstrapped dataset. Unless other sources will later also be mapped to the same attribute

Reply:
You are right. I forgot that the hash key for the tokens contains the attribute name. I will change it to an existing attribute, and the user can specify whichever attribute that makes the most sense for the project.